Added the possibility to specify resolvers from the repl

etorreborre opened the pull request:
I don't know if this is what you had in mind but this works ok for me. Feel free to change, adjust, etc...

When I ran the repl tests, I had contradictory messages like:
```
[info] Tests: 166
[info] Passed: 166
[info] Failed: 0
[error] (ops/test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 632 s, completed 10/12/2015 10:16:55 PM
```

Maybe you can help me there?

lihaoyi commented:
Looks great. Not sure why the test framework is misbehaving, let's see what CI says

lihaoyi commented:
CI seems unhappy. Looks like the test you added failed

https://travis-ci.org/lihaoyi/Ammonite/jobs/96009323#L1661

Seems like it can't find the dependency. No idea why

And your lines are too long :stuck_out_tongue: 

https://travis-ci.org/lihaoyi/Ammonite/jobs/96009323#L622

lihaoyi commented:
Now it looks broken in 2.10. Is that library not published for 2.10? We could just skip the test if that's the case

lihaoyi commented on build.sbt:
```diff
@@ -78,6 +78,10 @@ lazy val ops = project
     name := "ammonite-ops"
   )
 
+lazy val prompt = shellPrompt in ThisBuild := { state =>
```
What's this about?

etorreborre replied:
This is about getting a better prompt for submodules inside sbt.

lihaoyi replied:
I don't understand =/

lihaoyi replied:
What prompt are you talking about? Why do we care about the submodule's prompt? Why is this prompt better?

etorreborre replied:
Sorry I should have been clearer. When you have a multi-module sbt project you can "navigate" from the top-level module to a submodule and only run the tests for that module:
```
./sbt
> project repl
> test
```

With this setting you will see the module name before the prompt so you know where you are
```
./sbt
> project repl
repl> test
```

etorreborre commented:
Cool, I better understand how this works now :-). I have added the possibility to specify resolvers using ivy-style patterns (I think my previous manual test might just have worked because the jar was already cached).

I also added 2 small sbt improvements:

 1. make utest the only test framework to avoid ScalaCheck intercepting arguments on the command line and reporting errors

 2. add a prompt with the module name when navigating to a given submodule, like `repl>`

etorreborre commented:
Yes the build is still broken. I changed the jar version to something that is published for both `2.10` and `2.11`.

etorreborre commented:
@lihaoyi [This 2.11.4 build](https://travis-ci.org/lihaoyi/Ammonite/jobs/96175902) is failing, do you have any idea why? It looks like it has been failing for quite some time.

lihaoyi commented:
Yeah that one's been misbehaving and i haven't had time to investigate =/ If the rest are green it should be fine

etorreborre commented:
Yes the rest is fine.

lihaoyi commented:
Looks good

lihaoyi commented:
@etorreborre this should be in 0.5.2, which contains this with some tweaks to make it properly invalidate ammonite's own ivy cache when you change resolvers 

http://lihaoyi.github.io/Ammonite/#0.5.2

Try it out and let me know if it works for you!
